sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -63,10 +63,27 @@ object PhysicalOperation extends PredicateHelper {
         val substitutedFields = fields.map(substitute(aliases)).asInstanceOf[Seq[NamedExpression]]
         (Some(substitutedFields), filters, other, collectAliases(substitutedFields))
 
-      case Filter(condition, child) if condition.deterministic =>
-        val (fields, filters, other, aliases) = collectProjectsAndFilters(child)
-        val substitutedCondition = substitute(aliases)(condition)
-        (fields, filters ++ splitConjunctivePredicates(substitutedCondition), other, aliases)
+      case filter: Filter if filter.condition.deterministic || filter.condition.isInstanceOf[And] =>
```

Comment:
`PhysicalOperation` is used in many places, so have you checked this change has no side-effect for the other behivours?

Reply:
Yes, I have checked this change and I think that it has no side-effect for the other behaviors.